fetch/layers.go

```diff
@@ -278,6 +278,19 @@ func (l *Logic) receiveLayerContent(ctx context.Context, layerID types.LayerID,
 		}
 	}
 
+	// process the certificate after blocks are fetched
+	if peerRes.data != nil && peerRes.data.HareOutput != nil {
+		cert := peerRes.data.HareOutput
+		if cert.BlockID != types.EmptyBlockID && !contains(peerRes.data.Blocks, cert.BlockID) {
+			logger.With().Warning("certificate block id not in peer's layer content")
+			peerRes.err = errInvalidCertificate
+		}
```

Comment:
We can't error with errInvalidCertificate in this case. If we receive a valid certificate for a BlockID that we can't find anywhere, then we need to try to find it from a different peer, or try again on next sync.